### PR TITLE
Test vm mock

### PR DESCRIPTION
**Pull request by Thykof**
related to https://github.com/massalabs/massa-sc-toolkit/issues/70 

I moved the vm-mock tests from the toolkit to the sdk

### CODE REVIEW

**Review by SebastienValla — COMMENTED** [review]
Nice ! I think we should create an issue to add all the others function

File: assembly/__tests__/vm-mock.spec.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    1, 2, 3,
+    3, 4, 7,
+    4, 5, 9,
+  ]);
+
+  // prettier-ignore
+  checksForEachLineThatThe('`greater than` of two integers', 'arg0 > arg1', isFalse, [
+    0, 1,
+    2, 3,
+  ]);
```
Comment (peterjah) [inline assembly/__tests__/vm-mock.spec.ts:93]: how is this related to vm-mock ?

**Comment by Thykof** [comment]
you are right: https://github.com/massalabs/massa-as-sdk/issues/116

**Review by peterjah — APPROVED** [review]

File: assembly/__tests__/vm-mock.spec.ts (hunk 1 above)
Comment (Thykof) [reply assembly/__tests__/vm-mock.spec.ts:93]: it's not, do we need this? if yes, where to put it?

**Review by SebastienValla — APPROVED** [review]

File: assembly/__tests__/vm-mock.spec.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+const keyTestArgs = new Args().add("test");
+const valueTestArgs = new Args().add("value");
+
+const keyTest = toBytes("test");
+const valueTest = toBytes("value");
+
+describe("Testing mocked Storage and CallStack", () => {
+  test("Testing the callstack", () => {
+    const callStack = Context.addressStack();
```
Comment (gregLibert) [inline assembly/__tests__/vm-mock.spec.ts:23]:
```suggestion
    // By convention addressStack returns always the two addresses.
    const callStack = Context.addressStack();
```

File: assembly/__tests__/vm-mock.spec.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  test("Testing the Storage setOf", () => {
+    Storage.setOf(testAddress, keyTest, valueTest);
+    expect(
+      fromBytes(
+        Storage.getOf(
+          new Address("A12E6N5BFAdC2wyiBV6VJjqkWhpz1kLVp2XpbRdSnL1mKjCWT6oR"),
+          toBytes("test")
+        )
+      )
+    ).toBe("value");
```
Comment (gregLibert) [inline assembly/__tests__/vm-mock.spec.ts:41]: You should use string or bytes, here your mixing the two and then the test is less clear...

File: assembly/__tests__/vm-mock.spec.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      Storage.getOf(
+        new Address("A12E6N5BFAdC2wyiBV6VJjqkWhpz1kLVp2XpbRdSnL1mKjCWT6oR"),
+        new Args().add("test")
+      )
+        .nextString()
+        .unwrap()
+    ).toBe("value");
+  });
+
+  test("Testing the Storage set", () => {
```
Comment (gregLibert) [inline assembly/__tests__/vm-mock.spec.ts:56]: I think you are testing the get function

File: assembly/__tests__/vm-mock.spec.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    Storage.set(keyTest, valueTest);
+    expect(fromBytes(Storage.get(toBytes("test")))).toBe("value");
+  });
+
+  test("Testing the Storage set", () => {
```
Comment (gregLibert) [inline assembly/__tests__/vm-mock.spec.ts:61]: I think you are testing the has function

File: assembly/__tests__/vm-mock.spec.ts [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    Storage.set(keyTest, valueTest);
+    expect(Storage.has(toBytes("test"))).toBeTruthy();
+  });
+
+  test("Testing the Storage set", () => {
```
Comment (gregLibert) [inline assembly/__tests__/vm-mock.spec.ts:66]: wrong test name here and maybe below...

File: assembly/__tests__/vm-mock.spec.ts [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    Storage.set(keyTest, valueTest);
+    expect(fromBytes(Storage.get(toBytes("test")))).toBe("value");
+  });
+
+  test("Testing the Storage set", () => {
+    Storage.set(keyTest, valueTest);
+    expect(Storage.has(toBytes("test"))).toBeTruthy();
```
Comment (gregLibert) [inline assembly/__tests__/vm-mock.spec.ts:63]: Why not using keyTest instead of toBytes("test") ?

File: assembly/__tests__/vm-mock.spec.ts [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+      )
+    ).toBeTruthy();
+  });
+
+  test("Testing event", () => {
+    generateEvent("I'm an event ");
+  });
+});
+
+describe("Test Table tests", () => {
```
Comment (gregLibert) [inline assembly/__tests__/vm-mock.spec.ts:81]: This is a code sample on how to do test table. Shall be removed from wm-mock

**Review by gregLibert — CHANGES_REQUESTED** [review]
:book: readibility and `test as doc` approach to improve.

**Review by gregLibert — APPROVED** [review]
:clap:

**Comment by Thykof** [comment]
# QA

see that the github action workflow is green: https://github.com/massalabs/massa-as-sdk/actions/runs/3756970386
